kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -1060,10 +1077,10 @@ class ResidualBasedNewtonRaphsonStrategy
     ///@name Member Variables
     ///@{
 
-    typename TLinearSolver::Pointer mpLinearSolver; /// The pointer to the linear solver considered
-    typename TSchemeType::Pointer mpScheme; /// The pointer to the time scheme employed
-    typename TBuilderAndSolverType::Pointer mpBuilderAndSolver; /// The pointer to the builder and solver employed
-    typename TConvergenceCriteriaType::Pointer mpConvergenceCriteria; /// The pointer to the convergence criteria employed
+    typename TLinearSolver::Pointer mpLinearSolver = nullptr; /// The pointer to the linear solver considered. WARNING: The B&S has one, and here another!! Check this
```

Comment:
I would say it is better to get a reference to the linear solver when needed, but as i recall there was some detail preventing it.

I would be favorable to only using a reference in the strategy and to have the builder and solver to own it

Reply:
We can add a method like GetLinearSolver which calls the B&S LinearSolver. Otherwise this can lead to two different linear solvers

Reply:
yes, i would agree about that

Reply:
I have done that way

Reply:
but then why do you need the pointer? you can completely take it out...

Reply:
I mean, you don't need the linear solver pointer to be a member of the class

Reply:
See https://github.com/KratosMultiphysics/Kratos/pull/6857